File: php/testcode/weakcipher_014.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_weakcipher_chacha20
function weakcipher014(BenchmarkRequest $req): BenchmarkResponse {
    $data = $req->post('data');
    $key = getenv('ENCRYPTION_KEY');
    $iv = random_bytes(12);
    $tag = '';
    $encrypted = openssl_encrypt($data, 'chacha20-poly1305', $key, 0, $iv, $tag); // vuln-code-snippet safe-line php_weakcipher_chacha20
    return BenchmarkResponse::json(['ciphertext' => base64_encode($iv . $tag . $encrypted)]);
}
// vuln-code-snippet end php_weakcipher_chacha20

File: ruby/testcode/weakcipher_014.rb
require_relative 'shared'
require 'openssl'

# vuln-code-snippet start ruby_weakcipher_env_key
def encrypt_env_key(req)
  plaintext = req.body_str
  key = [ENV.fetch('ENCRYPTION_KEY')].pack('H*') # vuln-code-snippet safe-line ruby_weakcipher_env_key
  cipher = OpenSSL::Cipher.new('AES-256-GCM')
  cipher.encrypt
  cipher.key = key
  nonce = cipher.random_iv
  cipher.auth_data = ''
  ciphertext = cipher.update(plaintext) + cipher.final
  tag = cipher.auth_tag
  BenchmarkResponse.ok({ nonce: nonce.unpack1('H*'), tag: tag.unpack1('H*'), data: ciphertext.unpack1('H*') }.to_s)
end
# vuln-code-snippet end ruby_weakcipher_env_key
